packages/contracts/controller/src/account.cairo

```diff
@@ -454,10 +504,21 @@ mod CartridgeAccount {
         fn assert_valid_span_signature(
             self: @ContractState, hash: felt252, signer_signatures: Array<SignerSignature>
         ) {
-            assert(signer_signatures.len() <= 2, 'invalid-signature-length');
+            assert(signer_signatures.len() >= 1, 'empty-signatures');
             assert(
                 self.is_valid_owner_signature(hash, *signer_signatures.at(0)), 'invalid-owner-sig'
             );
+            match self.guardian_guid.read() {
+                Option::Some(_) => {
+                    assert(signer_signatures.len() >= 2, 'no-guardian-sig');
```

Comment:
when are there more than 2 sigs?

Reply:
it's a sanity check. i think it's good it's here. we want to make sure the user doesn't pass more than two signatures since there is no reason to do so. 
Argent's imlementation also checks there are no redundant signatures
https://github.com/argentlabs/argent-contracts-starknet/blob/1352198956f36fb35fa544c4e46a3507a3ec20e3/src/presets/argent_account.cairo#L963

Reply:
in this case assert will pass if there are 3 signatures though

Reply:
Line 514 asserts that, i wanted to provide separate error messages in each case